geomstats/geometry/invariant_metric.py

```diff
@@ -7,13 +7,81 @@
 from geomstats.geometry.matrices import Matrices
 from geomstats.geometry.riemannian_metric import RiemannianMetric
 from geomstats.integrator import integrate
-from geomstats.numerics.geodesic import ExpODESolver, LogShootingSolver
-from geomstats.numerics.ivp import GSIVPIntegrator
+from geomstats.numerics.bvp import ScipySolveBVP
+from geomstats.numerics.geodesic import ExpODESolver, LogODESolver
+from geomstats.numerics.ivp import ScipySolveIVP
 from geomstats.vectorization import repeat_out
 
 EPSILON = 1e-6
 
 
+class InvariantMetricMatrixExpODESolver(ExpODESolver):
+    """An exp solver adapted to _InvariantMetricMatrix."""
+
+    def exp(self, space, tangent_vec, base_point):
+        r"""Compute Riemannian exponential of tan. vector wrt to base point.
+
+        If :math:`\gamma` is a geodesic, then it satisfies the
+        Euler-Poincare equation [Kolev]_:
+
+        .. math::
+
+            \dot{\gamma}(t) = (dL_{\gamma(t)}) X(t)
+            \dot{X}(t) = ad^*_{X(t)}X(t)
+
+        where :math:`ad^*` is the dual adjoint map with respect to the
+        metric. For a right-invariant metric, :math:`dR` is used instead of
+        :math:`dL` and :math:`ad^*` is replaced by :math:`-ad^*`. The
+        exponential map is approximated by numerical integration
+        of this equation, with initial conditions :math:`\dot{\gamma}(0)`
+        given by the argument `tangent_vec` and :math:`\gamma(0)` by
+        `base_point`.
+
+        Parameters
+        ----------
+        tangent_vec : array-like, shape=[..., n, n]
+            Tangent vector at a base point.
+        base_point : array-like, shape=[..., n, n]
+            Point in the group.
+            Optional, defaults to identity if None.
+
+        Returns
+        -------
+        exp : array-like, shape=[..., n, n]
+            Point in the group equal to the Riemannian exponential
+            of tangent_vec at the base point.
+
+        References
+        ----------
+        https://en.wikipedia.org/wiki/Runge–Kutta_methods
+
+        .. [Kolev]   Kolev, Boris. “Lie Groups and Mechanics: An Introduction.”
+                     Journal of Nonlinear Mathematical Physics 11, no. 4, 2004:
+                     480–98. https://doi.org/10.2991/jnmp.2004.11.4.5.
+        """
+        base_point, left_angular_vel = space.metric._pre_exp(tangent_vec, base_point)
+        return super().exp(space, left_angular_vel, base_point)
+
+    def geodesic_ivp(self, space, tangent_vec, base_point):
+        """Geodesic curve for initial value problem."""
+        base_point, left_angular_vel = space.metric._pre_exp(tangent_vec, base_point)
+        return super().geodesic_ivp(space, left_angular_vel, base_point)
+
+
+class InvariantMetricMatrixLogODESolver(LogODESolver):
+    """A log solver adapted to _InvariantMetricMatrix."""
+
+    def log(self, space, point, base_point):
+        """Logarithm map."""
```

Comment:
Docstring missing: can you explain why the super().log is not enough for this log function?
Also, which is the space.metric.left (left-metric) hardcoded: does it only work for the left invariant metrics? If so, it should be indicated in the docstring?

^^^Could you put these explanations in the docstrings?

Reply:
Relates with above answer: `geodesic_equation` is written in the `LieAlgebra`, we are finding a solution that lives on the `LieAlgebra`, we need to translate it to the `LieGroup`.

Notice `left` is not hard-coded. It is a boolean that if `True` means "left", else "right". This boolean is controlled at instantiation time.